docs/ldml/tr35-keyboards.md

```diff
@@ -1348,206 +1342,615 @@ This example shows some of the mappings for a French keyboard layout:
+
+    - …
+    - marker `circ_marker`
+
+3. Also due to the `<display>` element, implementations can opt to display a visible `^` (perhaps visually distinct from a plain `^` carat). Implementations may opt to display nothing and only store the marker in the input context.
+
+4. The user now presses the `e` key, which is also added to the input context. The input context now has:
+
+    - …
+    - character `e`
+    - marker `circ_marker`
+
+5. Now, the input context matches the transform.  The `e` and the marker are replaced with `ê`.
+
+    The input context now has:
+
+    - …
+    - character `ê`
+
+**Effect of markers on final text**
+
+All markers must be removed before text is returned to the application from the input context.
+If the input context changes, such as if the cursor or mouse moves the insertion point somewhere else, all markers in the input context are removed.
+
+**Implementation Notes**
+
+Ideally, markers are implemented entirely out-of-band from the normal text stream. However, implementations _may_ choose to map each marker to a [Unicode private-use character](https://www.unicode.org/glossary/#private_use_character) for use only within the implementation’s processing and temporary storage in the input context.
+
+For example, the first marker encountered could be represented as U+E000, the second by U+E001 and so on.  If a regex processing engine were used, then those PUA characters could be processed through the existing regex processing engine.  `[^\uE000-\uE009]` could be used as an expression to match a character that is not a marker, and `[Ee]\u{E000}` could match `E` or `e` followed by the first marker.
+
+Such implementations must take care to remove all such markers (see prior section) from the resultant text. As well, implementations must take care to avoid conflicts if applications themselves are using PUA characters, such as is often done with not-yet-encoded scripts or characters.
+
+* * *
+
+### <a name="Element_transformGroup" href="#Element_transformGroup">Element: transformGroup</a>
+
+> <small>
+>
+> Parents: [transforms](#Element_transforms)
+>
+> Children: [import](#Element_import), [reorder](#Element_reorder), [_special_](tr35.md#special), [transform](#Element_transform)
+>
+> Occurrence: optional, multiple
+> </small>
+
+A `transformGroup` represents a set of transform elements or reorder elements.
+
+Each `transformGroup` is processed entirely before proceeding to the next one.
+
+
+Each `transformGroup` element, after imports are processed, must have either [reorder](#Element_reorder) elements or [transform](#Element_transform) elements, but not both. The `<transformGroup>` element may not be empty.
+
+**Examples**
+
+#### Example: `transformGroup` with only `import` elements
+
+This is a `transformGroup` that consists only of [`import`](#element-import) elements.  With this form, only `transform` OR `reorder` elements may be imported by all import statements, but not both.
```

Comment:
I'm still not seeing how this isn't just a special case of both of the other two `transformGroup` types.

Both of the other two state:
> This is a `transformGroup` that consists of one or more _____ elements, prefaced by one or more `import` elements. See the discussion of those elements for details.

(Fill in the blank with either `transform` or `reorder`.)

If the key point is that even without rules, only one type is supported... I think that'd be clearer if this example followed the other two entries, as they'd make the pattern clearer, I think.

See my related suggestions for the definition of the other two groups.  Hopefully that'll help to clarify my issue here.  Those should make the fact that _this_ group is basically a strict subset of the other two clearer.  (Yes, this one doesn't specify either a `transform` or a `reorder` element directly, but it does by proxy via `import`.)

Reply:
it's a leftover that should just be removed…